syne_tune/optimizer/schedulers/neuralbands/neuralband.py

```diff
@@ -0,0 +1,301 @@
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+import copy
+import logging
+import os
+from typing import Dict, Optional, List
+import numpy as np
+
+from syne_tune.backend.trial_status import Trial
+from syne_tune.optimizer.scheduler import SchedulerDecision
+from syne_tune.optimizer.schedulers.fifo import FIFOScheduler
+from syne_tune.optimizer.schedulers.hyperband_cost_promotion import (
+    CostPromotionRungSystem,
+)
+from syne_tune.optimizer.schedulers.hyperband_pasha import PASHARungSystem
+from syne_tune.optimizer.schedulers.hyperband_promotion import PromotionRungSystem
+from syne_tune.optimizer.schedulers.hyperband_rush import (
+    RUSHPromotionRungSystem,
+    RUSHStoppingRungSystem,
+)
+from syne_tune.optimizer.schedulers.hyperband_stopping import StoppingRungSystem
+from syne_tune.optimizer.schedulers.searchers.utils.default_arguments import (
+    check_and_merge_defaults,
+    Integer,
+    Boolean,
+    Categorical,
+    filter_by_key,
+    String,
+    Dictionary,
+    Float,
+)
+
+from syne_tune.optimizer.schedulers.searchers.searcher import BaseSearcher
+from syne_tune.optimizer.schedulers.searchers.searcher_factory import searcher_factory
+
+logger = logging.getLogger(__name__)
+
+def is_continue_decision(trial_decision: str) -> bool:
+    return trial_decision == SchedulerDecision.CONTINUE
+
+from syne_tune.optimizer.schedulers.hyperband import HyperbandBracketManager, _ARGUMENT_KEYS, _CONSTRAINTS
+from syne_tune.optimizer.scheduler import (
+    TrialScheduler,
+    SchedulerDecision,
+    TrialSuggestion,
+)
+
+from syne_tune.optimizer.schedulers.hyperband import _get_rung_levels, _is_positive_int, _sample_bracket
+from syne_tune.config_space import cast_config_values
+from syne_tune.optimizer.schedulers.searchers.bayesopt.datatypes.hp_ranges import (
+    HyperparameterRanges,
+)
+from syne_tune.optimizer.schedulers.searchers.bayesopt.datatypes.hp_ranges_factory import (
+     make_hyperparameter_ranges,
+)
+from syne_tune.optimizer.schedulers.neuralbands.networks import Exploitation
+from syne_tune.optimizer.schedulers.hyperband import HyperbandScheduler
+from syne_tune.optimizer.schedulers.neuralbands.neuralband_supplement import NeuralbandSchedulerBase
+
+
+class NeuralbandScheduler(NeuralbandSchedulerBase):
+    def __init__(self, config_space, gamma = 0.01, nu = 0.01, step_size = 30, max_while_loop = 100,  **kwargs):
+        """
+        NeuralBand is a neural-bandit-based HPO algorithm under multi-fedility setting, where a budget-aware neural network is
+        carefully designed and the feedback perturbation is utilized for efficient fidelity-aware exploration.  
```

Comment:
'NeuralBand is a neural-bandit-based HPO algorithm under multi-fedility setting, where a budget-aware neural network is carefully designed and the feedback perturbation is utilized for efficient fidelity-aware exploration. ' -> 'NeuralBand is a neural-bandit based HPO algorithm for the multi-fidelity setting. It uses a budget-aware neural network together with a feedback perturbation to efficiently explore the input space across fidelities.

Reply:
Done.